product_net_weight/views/view_product_template.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!--
+Copyright (C) 2021 - Today: GRAP (http://www.grap.coop)
+@author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+-->
+<odoo>
+
+    <record id="view_product_template_form" model="ir.ui.view">
+        <field name="model">product.template</field>
+        <field
+            name="inherit_id"
+            ref="stock.product_template_form_view_procurement_button"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='responsible_id']" position="after">
+                <label
+                    for="net_weight"
+                    invisible="product_variant_count > 1 and not is_product_variant"
+                />
+                <div
+                    class="o_row"
+                    name="net_weight"
+                    invisible="product_variant_count > 1 and not is_product_variant"
+                >
+                    <field name="net_weight" class="oe_inline" />
+                    <span><field name="weight_uom_name" /></span>
```

Comment:
Would recommend to remove the <span> elements to align units with the standard product form

Reply:
@drewes a new PR where the issue is fixed https://github.com/OCA/product-attribute/pull/1735/files